flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java

```diff
@@ -108,6 +111,11 @@
 
 	private final Options ALL_OPTIONS;
 
+	private static final String fileName = "yarn-app.ini";
+	private static final String cookieKey = "secureCookie";
```

Comment:
I think the ini file format is actually fine. Could also be JSON but I don't mind. To not break backwards-compatibility, I think we have to keep the behavior to use the last-used application id in case none is supplied. We could have an extra config entry for that.